bokehjs/src/coffee/models/scales/linear_scale.coffee

```diff
@@ -4,21 +4,20 @@ import * as p from "core/properties"
 
 export class LinearScale extends Scale
 
-  connect_signals: () ->
-    super()
+  @getters {
+    state: () ->
+      if not @_state? then @_state = @_compute_state()
+      return @_state
+  }
+
+  connect_range_signals: () ->
```

Comment:
All of this signal connection stuff serves to merely cache the Scale state. I'm not sure there's much (if any) performance benefit to be gains from the indirection. We could do away with the getter and signals and instead call `_compute_state` every time `compute` or `invert` is called.